lib/routes/account.js

```diff
@@ -891,12 +893,37 @@ module.exports = function (
       handler: function (request, reply) {
         log.begin('Account.RecoveryEmailStatus', request)
         var sessionToken = request.auth.credentials
-        reply(
-          {
-            email: sessionToken.email,
-            verified: sessionToken.emailVerified
+
+        cleanUpIfAccountInvalid()
+          .then(createResponse)
+          .done(reply, reply)
+
+        function createResponse() {
+          return {
+             email: sessionToken.email,
+             verified: sessionToken.emailVerified
+           }
+        }
+
+        function cleanUpIfAccountInvalid() {
```

Comment:
Extremely minor nit: as I read this, I expected `cleanUpIfAccountInvalid` to come before `createResponse` because that's the order they're invoked in.